src/setuptools_scm/hg_git.py

```diff
@@ -80,54 +90,56 @@ def node(self):
 
         return git_node[:7]
 
-    def count_all_nodes(self):
+    def count_all_nodes(self) -> int:
         revs, _, _ = self.do_ex("hg log -r 'ancestors(.)' -T '.'")
         return len(revs)
 
-    def default_describe(self):
+    def default_describe(self) -> _t.CmdResult:
         """
         Tentative to reproduce the output of
 
         `git describe --dirty --tags --long --match *[0-9]*`
 
         """
-        hg_tags, _, ret = self.do_ex(
+        hg_tags_str, _, ret = self.do_ex(
             [
                 "hg",
                 "log",
                 "-r",
-                "(reverse(ancestors(.)) and tag(r're:[0-9]'))",
+                "(reverse(ancestors(.)) and tag(r're:v?[0-9].*'))",
                 "-T",
                 "{tags}{if(tags, ' ', '')}",
             ]
         )
         if ret:
-            return None, None, None
-        hg_tags = hg_tags.split()
+            return _FAKE_GIT_DESCRIBE_ERROR
+        hg_tags: set[str] = set(hg_tags_str.split())
```

Comment:
The order is important. One cannot use a set.

`hg_tags: list[str] = hg_tags_str.split()`

Reply:
FYI, I've also seen people use a `dict[str, None]` for an ordered set, if you actually need the set behavior. Not ideal, but does work and is pretty fast. (Assuming you don't and list is better, but just pointing it out)